Bluewire.Conventions.UnitTests/VersionSemanticsTests.cs

```diff
@@ -0,0 +1,80 @@
+            // But not non-zero builds of previous major.minors.
+            new IsAncestorCase { Reference = "20.21.20-alpha.g35897262", Subject = "20.20.10-beta", Expected = false },
+            new IsAncestorCase { Reference = "20.21.20-canary", Subject = "20.20.10-beta", Expected = false },
+            // Unless we know they're in the ancestry of subsequent major.minors.
+            new IsAncestorCase { Reference = "20.21.20-alpha.g35897262", Subject = "20.20.10-beta", LastSubjectMasterBuild = 100, Expected = true },
+            new IsAncestorCase { Reference = "20.21.20-canary", Subject = "20.20.10-beta", LastSubjectMasterBuild = 100, Expected = true },
+            new IsAncestorCase { Reference = "20.21.20-beta", Subject = "20.20.10-beta", LastSubjectMasterBuild = 100, Expected = true },
+            new IsAncestorCase { Reference = "20.21.20-rc", Subject = "20.20.10-beta", LastSubjectMasterBuild = 100, Expected = true },
+            new IsAncestorCase { Reference = "20.21.20-release", Subject = "20.20.10-beta", LastSubjectMasterBuild = 100, Expected = true },
+
+            // Release/candidate/etc branches have significantly earlier build of weaker tag as an ancestor.
+            new IsAncestorCase { Reference = "20.21.20-rc", Subject = "20.21.10-beta", Expected = true },
+            new IsAncestorCase { Reference = "20.21.20-release", Subject = "20.21.10-rc", Expected = true },
+            // But not stronger tags.
+            new IsAncestorCase { Reference = "20.21.20-beta", Subject = "20.21.10-rc", Expected = false },
+            new IsAncestorCase { Reference = "20.21.20-rc", Subject = "20.21.10-release", Expected = false },
+            // Or later builds.
+            new IsAncestorCase { Reference = "20.21.10-rc", Subject = "20.21.20-beta", Expected = false },
+            new IsAncestorCase { Reference = "20.21.10-release", Subject = "20.21.20-rc", Expected = false },
+
+            // Weaker tags must be sufficiently earlier, because they are integrated via merges.
+            new IsAncestorCase { Reference = "20.21.10-rc", Subject = "20.21.10-beta", Expected = false },
+            new IsAncestorCase { Reference = "20.21.10-rc", Subject = "20.21.9-beta", Expected = true },
+            new IsAncestorCase { Reference = "20.21.10-rc", Subject = "20.21.8-beta", Expected = true },
+
+            new IsAncestorCase { Reference = "20.21.10-release", Subject = "20.21.10-rc", Expected = false },
+            new IsAncestorCase { Reference = "20.21.10-release", Subject = "20.21.9-rc", Expected = true },
+            new IsAncestorCase { Reference = "20.21.10-release", Subject = "20.21.8-rc", Expected = true },
+
+            new IsAncestorCase { Reference = "20.21.10-release", Subject = "20.21.10-beta", Expected = false },
+            new IsAncestorCase { Reference = "20.21.10-release", Subject = "20.21.9-beta", Expected = false },

```

Comment:
Think this should be Expected = true

Reply:
No, it shouldn't. A beta commit cannot reach a release branch directly: it must go via the candidate branch. Therefore a gap of at least two commits should exist.

Reply:
A key point here is that 'at least two' commits must exist... we cannot know how many must exist without more information, so comparing versions with different tags is actually completely meaningless. I'll remove this.